### PR TITLE
feat: add API to set color options to RichTextEditor

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Related to https://github.com/vaadin/web-components/pull/7392

## Type of change

- Enhancement

## Note

Currently `getColorOptions()` doesn't return the default value of the corresponding web component property.
This could be probably workarounded by adding `notify: true` and using `@Synchronize` annotation.

### CODE REVIEW

File: vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java [hunk 1]
```diff
@@ -269,6 +272,32 @@ private String getDeltaValue() {
...
+     * @since 24.5
+     * @return the list of colors options
+     */
+    public List<String> getColorOptions() {
+        return JsonSerializer.toObjects(String.class,
+                (JsonArray) getElement().getPropertyRaw("colorOptions"));
+    }
+
+    /**
+     * Sets the list of colors used by the text color picker and background
```
Comment (knoobie) [inline vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java:288]: What format is required here to be considered a color? hex, hsl, name.. something else? 😀

Comment (web-padawan) [reply vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java]: Added a mention that colors should be in HEX format (the web component popup uses that).
Note: the actual HTML value produced by Quill uses `rgb()` so the output looks like this:

```html
<p><span style="color: rgb(230, 0, 0);">Foo</span></p>
```

File: vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java [hunk 2]
```diff
@@ -268,6 +270,32 @@ private String getDeltaValue() {
         return getElement().getProperty("value");
     }
 
+    /**
+     * Gets the list of colors in HEX format used by the text color picker and
+     * background color picker controls of the text editor.
```
Comment (sissbruecker) [inline vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java:275]: Let's mention that this doesn't have a default value and returns null by default. Maybe also elaborate that in this case the web component shows a default palette.

Apart from that modifying the list returned by the method doesn't do anything. Let's return an unmodifyable list which would trigger a respective exception when trying to modify it, and mention in the JavaDoc that this returns an umnodifyable list. There are some similar use cases in the repo already (e.g. setting avatar items).

Comment (web-padawan) [reply vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java]: Thanks, updated.

**Pull request merged**